analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UriShouldNotBeHardcoded.CSharp12.cs

```diff
@@ -1,7 +1,44 @@
-﻿class PrimaryConstructor(string ctorParam = "file://blah.txt") // FN
+﻿using System;
+using System.Collections.Generic;
+
+class PrimaryConstructor(string ctorParamUri = "file://blah.txt", // Noncompliant
+                         string ctorParam = "file://blah.txt") // Compliant
 {
-    void Method(string methodParam = "file://blah.txt") // FN
+    void Method(string methodParamUri = "file://blah.txt", // Noncompliant
+                string methodParam = "file://blah.txt") // Compliant
     {
-        var lambda = (string lambdaParam = "file://blah.txt") => lambdaParam; // FN
+        var lambda = (string lambdaParamUri = "file://blah.txt") => lambdaParamUri; // Noncompliant
+        var lambda2 = (string lambdaParam = "file://blah.txt") => lambdaParam; // Compliant
+
+        var lambdaMultipleNoncompliances = (string paramUri1 = "file://blah.txt", string paramUri2 = "file://blah.txt") => paramUri1;
+        //                                                     ^^^^^^^^^^^^^^^^^
+        //                                                                                           ^^^^^^^^^^^^^^^^^@-1
+    }
+}
+
+struct PrimaryConstructorStruct(string ctorParamUri = "file://blah.txt", // Noncompliant
+                         string ctorParam = "file://blah.txt") // Compliant
```

Comment:
`string ctorParam...` is misaligned. Rather than trying to indent vertically with `string ctorParamUri`, I would indent of a single level (i.e. 4 spaces w.r.t. the definition of the `struct`).
```cs
struct PrimaryConstructorStruct(string ctorParamUri = "file://blah.txt", // Noncompliant
    string ctorParam = "file://blah.txt") // Compliant
{
}
```

Or, in my opinion, better, in the following way:
```cs
struct PrimaryConstructorStruct(
    string ctorParamUri = "file://blah.txt", // Noncompliant
    string ctorParam = "file://blah.txt") // Compliant
{
}
```

Both these solutions have the advantage that, if the name of the struct changes, the following lines of the signature don't get misaligned (which I think it's what has happened here).